src/generic/grid.cpp

```diff
@@ -6180,7 +6299,42 @@ void wxGrid::OnKeyDown( wxKeyEvent& event )
                 break;
 
             case WXK_ESCAPE:
-                ClearSelection();
+                if ( m_isDragging && m_winCapture )
+                {
+                    switch ( m_cursorMode )
+                    {
+                        case WXGRID_CURSOR_MOVE_COL:
+                        case WXGRID_CURSOR_MOVE_ROW:
+                            // end row/column moving
+                            m_winCapture->Refresh();
+                            m_dragLastPos = -1;
+                            break;
+
+                        case WXGRID_CURSOR_RESIZE_ROW:
+                        case WXGRID_CURSOR_RESIZE_COL:
+                            // reset to size from before dragging
+                            if ( m_cursorMode == WXGRID_CURSOR_RESIZE_ROW )
+                                wxGridRowOperations().SetLineSize(this,
+                                    m_dragRowOrCol, m_dragRowOrColOldSize);
+                            else
+                                wxGridColumnOperations().SetLineSize(this,
+                                    m_dragRowOrCol, m_dragRowOrColOldSize);
```

Comment:
```suggestion
                            (m_cursorMode == WXGRID_CURSOR_RESIZE_ROW
                                ? wxGridRowOperations()
                                : wxGridColumnOperations())
                            .SetLineSize(this,                                    m_dragRowOrCol, m_dragRowOrColOldSize);
```

Reply:
To make this work we need to cast the arguments of the ternary operator to the common base class explicitly. This is a bit ugly but arguably still worth avoiding duplicating `SetLineSize()` call, I think... but please let me know if you disagree.